packages/core/upload/admin/src/components/UploadAssetDialog/PendingAssetStep/PendingAssetStep.js

```diff
@@ -30,15 +31,28 @@ export const PendingAssetStep = ({
   onClickAddAsset,
   onCancelUpload,
   onUploadSucceed,
+  trackedLocation,
 }) => {
   const assetCountRef = useRef(0);
   const { formatMessage } = useIntl();
+  const { trackUsage } = useTracking();
   const [uploadStatus, setUploadStatus] = useState(Status.Idle);
 
   const handleSubmit = async e => {
     e.preventDefault();
     e.stopPropagation();
 
+    const assetsCountByType = assets.reduce(
+      (acc, cur) => {
+        const type = cur.type;
+        acc[type] += 1;
+
+        return acc;
+      },
+      { doc: 0, audio: 0, video: 0, image: 0 }
+    );
```

Comment:
Mh, I must say that I am not a fan of the default value, because if we add another type one day, this is hard to spot and to refactor. What do you think about building the object dynamically?

```suggestion
    const assetsCountByType = assets.reduce(
      (acc, asset) => {
        const { type } = asset;
        
        if (!acc[type]) {
          acc[type] = 0;
        }

        acc[type] += 1;

        return acc;
      },
      {}
    );
```